### PR TITLE
replace embedded kafka with testcontainers

### PR DESCRIPTION
**Pull request by bplommer**
Needed because of binary compatibility issues with scala 3

### CODE REVIEW

**Comment by bplommer** [comment]
@vlovgr this is now ready to review for real 🙂

File: license.txt [hunk 1]
```diff
@@ -174,3 +174,30 @@
       of your accepting any such warranty or additional liability.
 
    END OF TERMS AND CONDITIONS
+
+-----
+
+This project (specifically BaseKafkaSpec.scala) contains code derived from
```
Comment (vlovgr) [inline license.txt:180]: Could we add the license to `BaseKafkaSpec.scala` as well?

**Review by vlovgr — APPROVED** [review]

File: license.txt (hunk 1 above)
Comment (bplommer) [reply license.txt:180]: sure :+1:

Comment (bplommer) [reply license.txt:180]: Done - by the way, should we add copyright/license headers to the test files?

**Pull request merged**